lib/ts/BufferWriterFormat.cc

```diff
@@ -336,70 +382,100 @@ namespace bw_fmt
     width -= static_cast<int>(n);
     string_view digits{buff + sizeof(buff) - n, n};
 
-    // The idea here is the various pieces have all been assembled, the only difference
-    // is the order in which they are written to the output.
-    switch (spec._align) {
-    case BWFSpec::Align::LEFT:
+    if (spec._align == BWFSpec::Align::SIGN) { // custom for signed case because prefix and digits are seperated.
       if (neg)
         w.write(neg);
       if (prefix1) {
         w.write(prefix1);
         if (prefix2)
           w.write(prefix2);
       }
-      w.write(digits);
-      while (width-- > 0)
-        w.write(spec._fill);
-      break;
-    case BWFSpec::Align::RIGHT:
       while (width-- > 0)
         w.write(spec._fill);
-      if (neg)
-        w.write(neg);
-      if (prefix1) {
-        w.write(prefix1);
-        if (prefix2)
-          w.write(prefix2);
-      }
-      w.write(digits);
-      break;
-    case BWFSpec::Align::CENTER:
-      for (int i = width / 2; i > 0; --i)
-        w.write(spec._fill);
-      if (neg)
-        w.write(neg);
-      if (prefix1) {
-        w.write(prefix1);
-        if (prefix2)
-          w.write(prefix2);
-      }
       w.write(digits);
-      for (int i = (width + 1) / 2; i > 0; --i)
-        w.write(spec._fill);
-      break;
-    case BWFSpec::Align::SIGN:
-      if (neg)
-        w.write(neg);
-      if (prefix1) {
-        w.write(prefix1);
-        if (prefix2)
-          w.write(prefix2);
-      }
-      while (width-- > 0)
-        w.write(spec._fill);
-      w.write(digits);
-      break;
-    default:
-      if (neg)
-        w.write(neg);
-      if (prefix1) {
-        w.write(prefix1);
-        if (prefix2)
-          w.write(prefix2);
+    } else { // use generic Write_Aligned
```

Comment:
Did you test the performance impact of this change? Integer output must be as fast as possible and there are constraints on how much slower than `snprintf` it can be (no more than 50% slower, preferably less than 25% slower). Uncomment out (termporarily) the performance tests in `test_BufferWriterFormat.cc` and see what the data indicates. Before this, `bwformat` took about 82ms while `snprintf` was around 70ms.

Reply:
Results (somehow faster than `snprintf`):
02 optimization:
```bw.print() 104128329ns or 104ms
Preformatted: 75558516ns or 75ms
snprint Timing is 114764286ns or 114ms```

No Optimization:
```bw.print() 968016030ns or 968ms
Preformatted: 610290929ns or 610ms
snprint Timing is 123027682ns or 123ms```

Reply:
That is a bit odd, you might run it a few times. It had been close, but I also changed the test case to do an integer and a string_view. I'll have to play with it, but if it's actually faster then we'll do that.